hooks/useExchange.ts

```diff
@@ -133,16 +129,15 @@ const useExchange = ({
 	const ETHBalance = ETHBalanceQuery.isSuccess ? ETHBalanceQuery.data ?? zeroBN : null;
 
 	const synthsWalletBalancesQuery = useSynthsBalancesQuery(walletAddress);
-	const synthsWalletBalance = synthsWalletBalancesQuery.isSuccess
-		? synthsWalletBalancesQuery.data
-		: null;
+	const synthsWalletBalance = synthsWalletBalancesQuery.data ?? null;
 
 	const exchangeRatesQuery = useExchangeRatesQuery();
 
-	const oneInchQuery = useOneInchTokenList();
+	const tokenList = isL2 ? optimismOneInchTokenList.tokens : mainnetOneInchTokenList.tokens;
+	const oneInchTokensMap: any = isL2
```

Comment:
Is it possible to format these the same so we can keep a type?

Reply:
Since the data is in a static JSON file, TS generates a strict type. The real issue is that the baseCurrencyKey and quoteCurrencyKey are strings and not specific types (which makes sense, unless we want to generate their type based on the JSON we get from 1inch). Overriding the types here is a temporary solution.

Reply:
Ok I think we can instead change to `const oneInchTokensMap: Record<string, Token>` with token type taken from tokenLists/types, just might need to change `chainId` type to number from NetworkId

Reply:
Yeah. I tried this, but it didn't work. But since we're bringing back the old query, I think it doesn't really matter. I'll add types to the query though, so at least we get some type coverage.